examples/nlp/language_modeling/conf/config.yaml

```diff
@@ -0,0 +1,106 @@
+# BERT Pretraining
+trainer:
+  gpus: 1 # the number of gpus, 0 for CPU, or list with gpu indices
+  num_nodes: 1
+  max_epochs: 2 # the number of training epochs
+  max_steps: null # precedence over max_epochs
+  accumulate_grad_batches: 1 # accumulates grads every k batches
+  precision: 16 # 16 to use AMP 
+  amp_level: O1 # O1 or O2 if using AMP 
+  distributed_backend: ddp
+  gradient_clip_val: 0.0
+  row_log_interval: 1
+  val_check_interval: 1.0 # check once per epoch .25 for 4 times per epoch
+  checkpoint_callback: false # provided by exp_manager
+  logger: false # provided by exp_manager
+
+model:
+  pretrained_model_name: bert-base-uncased # huggingface model name
+  max_seq_length: 128
+  # The maximum total input sequence length after tokenization. Sequences longer than this
+  # will be truncated, and sequences shorter than this will be padded.
+  mask_prob: 0.15
+  # Probability of masking a token in the input text during data processing.
+  short_seq_prob: 0.1
+  # Probability of having a sequence shorter than the maximum sequence length `max_seq_length` in data processing.",
+  batch_size: 16 # per GPU
+
+  language_model:
+    pretrained_model_name: ${model.pretrained_model_name}
+    bert_checkpoint: null
+    bert_config:
+      attention_probs_dropout_prob: 0.1
+      hidden_act: gelu
+      hidden_dropout_prob: 0.1
+      hidden_size: 768
+      initializer_range: 0.02
+      intermediate_size: 3072
+      max_position_embeddings: 512
+      num_attention_heads: 12
+      num_hidden_layers: 12
+      type_vocab_size: 2
+      vocab_size: 30522
+    bert_config_file: null # json file, precedence over bert_config
+
+  tokenizer:
+    tokenizer_name: nemobert # or sentencepiece
+    pretrained_model_name: ${model.pretrained_model_name}
+    special_tokens: null
+    vocab_file: null # path to vocab file
+    data_file: null # used to build sentencepiece
+    tokenizer_model: null # tokenizer model for sentencepiece
+    sample_size: null # for sentencepiece
+    vocab_size: ${model.language_model.bert_config.vocab_size} # vocab size for building sentence piece
+    do_lower_case: null # only used when tokenizer is build with vocab file or with sentencepiece
+   
+  train_ds:
+    data_file: null # path to data file
+    max_seq_length: ${model.max_seq_length}
+    mask_prob: ${model.mask_prob}
+    short_seq_prob: ${model.short_seq_prob}
+    batch_size: ${model.batch_size}
+    shuffle: true
+    num_samples: -1
+    num_workers: 2
+    use_cache: true
+    drop_last: false
+    pin_memory: false
+
+  validation_ds:
+    data_file: null # path to data file
+    max_seq_length: ${model.max_seq_length}
+    mask_prob: ${model.mask_prob}
+    short_seq_prob: ${model.short_seq_prob}
+    batch_size: ${model.batch_size}
+    shuffle: false
```

Comment:
and here?

Reply:
We can have the user input this information once and then use it multiple times.